gmailr_pull.Rmd

```diff
@@ -65,11 +51,32 @@ get_info_robust <- function(id) {
                to_name = NA, to_email = NA, to_domain = NA, from_domain = NA)
   })
 }
+```
+
+```{r}
+# get message ids
+n <- 100
+system.time({m <- messages(user_id = 'me', num_results = n) %>% gmailr::id()})
+system.time({mes <- lapply(m, possibly(message, otherwise = NULL), format = "metadata")})
+system.time({
+  res <- data_frame(
+    id = m[!map_lgl(mes, is.null)],
+    to = map_chr(mes, to),
+    from = map_chr(mes, from),
+    date = as.POSIXct(strptime(map_chr(mes, gmailr::date), "%a, %d %b %Y %X %z")),
```

Comment:
note to self: check if this date conversion corrects for the time zone ...
it was very important to get everything to UTC (hence this line):

```r
     mutate(date = if_else(tz_sign == "+", date - hours(tz_magn), date + hours(tz_magn))) %>%		
```